lib/ansible/modules/cloud/ovirt/ovirt_templates.py

```diff
@@ -304,7 +312,47 @@ def main():
                     'id': template.id,
                     'template': get_dict_of_struct(template),
                 }
+        elif state == 'registered':
+            storage_domains_service = connection.system_service().storage_domains_service()
+
+            # Find the storage domain with unregistered template:
+            sd = search_by_name(storage_domains_service, module.params['storage_domain'])
+            if sd is None:
+                raise ValueError(
+                    "Storage domain '%s' wasn't found." % module.params['storage_domain']
+                )
+            storage_domain_service = storage_domains_service.storage_domain_service(sd.id)
+            templates_service = storage_domain_service.templates_service()
+
+            # Find the the unregistered Template we want to register:
+            templates = templates_service.list(unregistered=True)
+            template = next(
+                (t for t in templates if t.name == module.params['name']),
+                None
+            )
+            if template is None:
```

Comment:
What if the template is already registered? We shouldn't fail, but rather exit with `changed=False`